Compat for Django 1.9 and 1.10

thomasgoirand opened the pull request:
Hi there!

I'm the package maintainer of python-jingo in Debian. I maintain it, because it's a build-dependency of python-django-compressor, which is itself used by OpenStack Horizon (the OpenStack dashboard), and I maintain all packages for OpenStack in Debian.

As we're going to ship Debian 9 (aka: Stretch, which we will freeze a the end of the year) with Django 1.10, we are already extensively testing and making sure that all of our packages are Django 1.10 ready. Therefore, we're testing them against the beta1 release of Django (which is currently in Debian Experimental).

I have therefore written a few patches for Jingo to support Django 1.10. As much as I can tell, you guys aren't even supporting Django 1.9, which is already old (more than 6 months). I have rebuilt the package in both Django 1.9 and 1.10 env, and all unit tests are passing. Well, all but:
- test_render_django
- test_render_django_no_toplevel_override
- test_render_django_toplevel_override

I haven't figured out yet how to fix these in Django 1.9 / 1.10, and I'm not much interested in it, because it looks like only fixing the unit tests themselves. However, if you can come with a patch, I'd love to use it.

Best would be if you could release a new version of Jingo with this pull request, plus the fix of the 3 unit tests above which are failing. In such case, please let me know (I'm zigo on Freenode and OFTC), and I'll make sure your new upstream release ends up in Debian Stretch.

Hoping this pull request helps,

Thomas Goirand (zigo)

thomasgoirand committed:
[Django 1.9] TemplateDoesNotExist removed

TemplateDoesNotExist is a private function that should never be used.
In Django 1.9, upstream authors decided to move it to another path.
This patch defines TemplateDoesNotExist directly in jingo instead of
using the internal Django version.

thomasgoirand committed:
[Django 1.9] run_tests.py must call django.setup()

This patch adds a call to django.setup() which is now needed in Django.

thomasgoirand committed:
[Django 1.9] Use six, but not from Django

Django 1.9 removed django.utils.six. Jingo then should use six
directly, not from Django.

thomasgoirand committed:
[Django 1.10] django.template.loader.BaseLoader removed

In Django 1.10, django.template.loader.BaseLoader is removed completely.
This patch uses django.template.loaders.base.Loader instead.

thomasgoirand committed:
[Django 1.10] add a few things to INSTALLED_APPS

In Django 1.10, we need to add this to INSTALLED_APPS:
    django.contrib.sites
    django.contrib.contenttypes
    django.contrib.auth

Without this, unit tests are failing over with not-found templates.

thomasgoirand committed:
[Django 1.10] defines TEMPLATES

In Django 1.10, many things which were previously declared separatly
now have to go in a TEMPLATES = [] object.

thomasgoirand committed:
[Django 1.10] Object settings.TEMPLATE_DEBUG doesn't exist

In Django 1.10, settings.TEMPLATE_DEBUG. Instead, we should use
settings.TEMPLATES[0]['OPTIONS']['debug'].

thomasgoirand committed:
[Django 1.10] django.forms.util renamed

In Django 1.10, django.forms.util was renamed django.forms.utils
(deprecation cycle ended, and there's no fallback).

thomasgoirand committed:
[Django 1.10] Loader method __init__() takes 2 args

In Django 1.10, the __init__() method of the Loader class takes
2 args, not 1 anymore. This patch fixes that.

thomasgoirand committed:
[Django 1.10] urls.patterns-removed

In Django 1.10, django.conf.urls.patterns (and its associated function)
doesn't exist anymore, and we *must* use url(), not just () for the
urlpatterns tuples.

thomasgoirand committed:
[Django 1.7] Retain django 1.7 compat

django.template.loaders.base doesn't exist in Django 1.7. This patch
makes the Django 1.10 to continue to work with older version.

thomasgoirand committed:
[Django 1.9] Also test with Django 1.9

jsocol commented:
@thomasgoirand as the only remaining maintainer of jingo, I've EOLed the project, as [noted in the README](https://github.com/jbalogh/jingo#jingo-is-deprecated). I've moved my projects to [django-jinja](https://github.com/niwinz/django-jinja) and recommend others do the same. I won't be doing any additional releases of jingo.

If you're interested in taking over maintenance of the project let me know, I can give you access to PyPI and we can direct people to your fork. Alternatively, you can always fork the project under a new name.

If you do want to pick up maintenance, I'd recommend only supporting currently-supported Django versions, i.e. 1.8–1.10. It helps a lot.

thomasgoirand committed:
[Django 1.9] As six as runtime requires

Previous commit uses six, not from Django. This patch adds six
as runtime requires.

thomasgoirand committed:
[Django < 1.10] Set the 2nd param of __init__ optional

Previous commit added a 2nd param for the __init__ of the Loader
class. However, this broke compat with older version of Django.
This patch solves that by making this 2nd parameter optional.

thomasgoirand committed:
Also add six as deps in tox.ini
